modules/edge-ren/src/lib.rs

```diff
@@ -0,0 +1,480 @@
+
+				Ok(())
+			})?;
+			Ok(())
+		}
+
+	}
+}
+
+
+impl<T: Config> Module<T> {
+
+	/// The account ID that holds the pallet's accumulated funds on pallet-assets; mostly fees for now, maybe for loss of exsistential deposit in future.
+    pub fn account_id() -> T::AccountId {
+        T::ModuleId::get().into_account()
+    }
+
+	/// Encode the values for creating the signature hash.
+	fn signable_message(p_hash: &[u8; 32], amount: BalanceOf<T>, to: &[u8], n_hash: &[u8; 32], token: &[u8; 32]) -> Vec<u8> {
+
+		let mut amount_slice = Encode::encode(&amount);
+		amount_slice.reverse();
+
+		// p_hash ++ amount ++ token ++ to ++ n_hash
+		let length = 32 + 32 + 32 + to.len() + 32;
+		let mut v = Vec::with_capacity(length);
+		v.extend_from_slice(&p_hash[..]);
+		v.extend_from_slice(&[0u8; 16][..]);
+		v.extend_from_slice(&amount_slice);
+		v.extend_from_slice(&token[..]);
+		v.extend_from_slice(to);
+		v.extend_from_slice(&n_hash[..]);
+		v
+	}
+
+	/// Verify that the signature has been signed by RenVM using the PublicKey of the token in token info
+	fn verify_signature(
+		_ren_token_id: TokenIdOf<T>,
+		p_hash: &[u8; 32],
+		amount: BalanceOf<T>,
+		to: &[u8],
+		n_hash: &[u8; 32],
+		sig: &[u8; 65],
+	) -> DispatchResult {
+
+		let ren_token = RenTokenRegistry::<T>::get(&_ren_token_id).ok_or_else(|| Error::<T>::RenTokenNotFound)?;
+		let identifier = ren_token.ren_token_renvm_id;
+
+		let signed_message_hash = keccak_256(&Self::signable_message(p_hash, amount, to, n_hash, &identifier));
+		let recoverd =
+			secp256k1_ecdsa_recover(&sig, &signed_message_hash).map_err(|_| Error::<T>::InvalidMintSignature)?;
+		let addr = &keccak_256(&recoverd)[12..];
+
+		let pub_key = ren_token.ren_token_pub_key;
+
+		ensure!(addr == pub_key, Error::<T>::InvalidMintSignature);
+
+		Ok(())
+	}
+
+}
+
+
+#[allow(deprecated)]
+impl<T: Config> frame_support::unsigned::ValidateUnsigned for Module<T> {
+	type Call = Call<T>;
+
+	/// The entry point for the validation process of unsigned transactions
+	fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+		if let Call::mint(_ren_token_id, who, p_hash, amount, n_hash, sig) = call {
+			// check if already exists
+			if Signatures::contains_key(&sig) {
+				return InvalidTransaction::Stale.into();
+			}
+
+			let verify_result = Encode::using_encoded(&who, |encoded| -> DispatchResult {
+				Self::verify_signature(*_ren_token_id, &p_hash, *amount, encoded, &n_hash, &sig.0)
+			});
+
+			// verify signature
+			if verify_result.is_err() {
+				return InvalidTransaction::BadProof.into();
+			}
+
+			ValidTransaction::with_tag_prefix("edge-ren")
+				.priority(T::RenvmBridgeUnsignedPriority::get())
+				.and_provides(sig)
+				.longevity(64_u64)
+				.propagate(true)
+				.build()
+		} else {
+			InvalidTransaction::Call.into()
+		}
+	}
+}
+
+
+/// Simple ensure origin for the RenVM account
+pub struct EnsureRenVM<T>(sp_std::marker::PhantomData<T>);
```

Comment:
Ultimately I don't see this origin being used anywhere and we should remove it unless we have a usecase. You are calling force_create_asset in your extended implementation of the asset pallet which doesn't have an origin parameter.

Reply:
Yes, currently we aren't using it and so we should remove it.